src/rockstor/system/smart.py

```diff
@@ -333,19 +345,57 @@ def available(device, custom_options="", test_mode=TESTMODE):
     :return: available (boolean), enabled (boolean)
     """
     if not test_mode:
-        o, e, rc = run_command(
-            [SMART, "--info"] + get_dev_options(device, custom_options)
-        )
+        """
+        Because Smart support is part of the nvme standard and smarctl --info
+        does not explicitly state if smart is supported or not, then smartctl --all
+        is used instead, and the presence of the smart status section is used to determine
+        if smart is indeed available. Some devices such as usb connected ssd may not give
+        this section
+        """
+        if "nvme" in device:
+            if  device in _failed_devices:   #smartctl has already failed before, so return without running it again
+                #logger.info("Device previously errored command smartctl --all %s",  device) #repeats every 30 secs unless disabled
+                return 0, 0  #Available and Enabled both False 
```

Comment:
```suggestion
                return False, False  #Available and Enabled both False 
```
@delboy711 Not a show-stopper as we look to be heading in the direction of a full refactor here. But I just wanted to point out that although this is equivalent, it could limit future type-hinting as 0 will be 'cast' as an integer, where-as True|False will be 'cast' as a boolean. But again, we are mid-refactor with these changes and I'd similarly like to throw out the existing rampant use of single char variables we have in this code.

Reply:
No worries committing this just yet - it was just to indicate a drawback in using equivalent values really. I'm mid-review currently and should be done shortly.